taichi/program/state_flow_graph.cpp

```diff
@@ -783,17 +916,16 @@ void StateFlowGraph::delete_nodes(
 
   nodes_ = std::move(new_nodes_);
   reid_nodes();
+  reid_pending_nodes();
```

Comment:
BTW, can we assert that all the nodes in `indices_to_delete` are pending?

Reply:
Maybe we can refactor `indices_to_delete` such that the `pending_node_id` instead of `node_id` is passed into the `unordered_map`?

Reply:
Ah yeah, that should be better :) I saw that you have explicitly converted a pending index into a "global" one when populating `indices_to_delete`..